File: playwright/po/loginPO.js
const { expect } = require('@playwright/test');

exports.loginPO = class loginPO {
  /**
  * @param {import('@playwright/test').Page} page
  */
  constructor(page) {
  this.page = page;
  this.Title_text = page.locator("xpath=//h1[text()='Login Page']");
  this.Username_textbox = page.locator("xpath=//input[@id='username']");
  this.Password_textbox = page.locator("xpath=//input[@id='password']");
  this.Login_button = page.locator("xpath=//button[text()='Login']");
  //Errors
  this.Error_InvalidUsername = page.locator("xpath=//b[text()='Your username is invalid!']");
  this.Error_InvalidPassword = page.locator("xpath=//b[text()='Your password is invalid!']");
  }
    
  async goto() {
    await this.page.goto('https://practice.expandtesting.com/login');
    }

  async VerifyExpectedElementsAppearOnLoginPage() {
    await expect(this.Title_text).toBeVisible();
    await expect(this.Username_textbox).toBeVisible();
    await expect(this.Password_textbox).toBeVisible();
    await expect(this.Login_button).toBeVisible();
    }
   
  async Login(userName, password) {
    await this.Username_textbox.fill(userName);
    await this.Password_textbox.fill(password);
    await this.Login_button.click(); 
  }

  async VerifyInvalidUsernameErrorAppears() {
    await expect(this.Error_InvalidUsername).toBeVisible()
    }
 
  async VerifyInvalidPasswordErrorAppears() {
    await expect(this.Error_InvalidPassword).toBeVisible()
    }      
  };

File: java-selenium/javaSelenium/src/main/java/pageObjects/loginPO.java
package pageObjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class loginPO {

	WebDriver driver;
	By LoginPage_Title_text=By.xpath("//h1[text()='Login Page']");
	By LoginPage_Username_textbox=By.xpath("//input[@id='username']");
	By LoginPage_Password_textbox=By.xpath("//input[@id='password']");
	By LoginPage_Login_button=By.xpath("//button[text()='Login']");
	//Errors
	By LoginPage_Error_InvalidUsername=By.xpath("//b[text()='Your username is invalid!']");
	By LoginPage_Error_InvalidPassword=By.xpath("//b[text()='Your password is invalid!']");
	
	public loginPO(WebDriver driver) {
		this.driver=driver;
	}
	
	public void VerifyExpectedElementsAppearOnLoginPage() {
		assertTrue(driver.findElement(LoginPage_Title_text).isDisplayed());
		assertTrue(driver.findElement(LoginPage_Username_textbox).isDisplayed());
		assertTrue(driver.findElement(LoginPage_Password_textbox).isDisplayed());
		assertTrue(driver.findElement(LoginPage_Login_button).isDisplayed());	
	}
	
	public void Login(String username, String password) {
		driver.findElement(LoginPage_Username_textbox).sendKeys(username);
		driver.findElement(LoginPage_Password_textbox).sendKeys(password);
		driver.findElement(LoginPage_Login_button).click();	
	}
	
	public void VerifyInvalidUsernameErrorAppears() {
		assertTrue(driver.findElement(LoginPage_Error_InvalidUsername).isDisplayed());
	}
	
	public void VerifyInvalidPasswordErrorAppears() {
		assertTrue(driver.findElement(LoginPage_Error_InvalidPassword).isDisplayed());
		}
}
